### DISCUSSION TITLE
Two independent prescribed velocities

### DISCUSSION THREAD

**Discussion by harryfor4** [Anura3D/Anura3D_OpenSource#56] [Q&A] [answered]
Hi Everyone,

I'm using Anura3D to simulate the pile installation problem with varying displacement rates during penetration. I want the pile to be driven at one penetration rate(PRESCRIBED VELOCITY 1) for some time in 1st stage of my problem and a different penetration rate(PRESCRIBED VELOCITY 2) in the next stage. In ANURA3D (2022 version), unlike "SOLID TRACTION", we cannot define two independent "Prescribed velocities". 

Query:
Can the final stress states of stage 1 be taken as the initial stress states for stage 2? 
Or
 Is there any way to deal with such kind of problems with ANURA3D?

**Comment by murphy-john** [comment, marked as answer]
1) Yes, the stresses at the material points are carried forward between load steps. The information is passed using binary (BRF) files. 

2) Edit: Alba corrected me, see [her comment](https://github.com/Anura3D/Anura3D_OpenSource/discussions/56#discussioncomment-4741016) on this tread. You _can_ vary the prescribed velocity. ~~I don't think the ability to automatically change PRESCRIBED VELOCITY like loading (e.g. solid traction, gravity, etc.) is in the current version. If the change occurs instantaneously, I believe you could run the first phase of your problem to a certain time (# load steps * time per load step). Then change the prescribed velocities in the CPS and/or GOM files and resume you model with for phase 2. You'd increase the number of loadsteps in the edited CPS file. If you want the velocity to change during a load step, I think you'll have to write a subroutine to do this. You can use the "solid traction" as a guide on how to do this.~~

**Reply by harryfor4** [reply]
Thank you. Both of your answers were informative and helped solve my problem.

**Comment by ayerrocolom** [comment]
I just want to complete the previous answer from @murphy-john .
In fact, the current version of Anura3D should allows for changing the prescribed velocity in the same manner as the solid traction or gravity using multipliers. These multipliers are defined in the CPS under $$PRESCRIBED_VELOCITY, and can be modified throughout the calculation as needed. For example, if the velocity defined in GiD (the one stored in the GOM file) is 2 cm/s and the multipliers in CPS are set to 1, the penetration will progress at 2 cm/s. If the multipliers in a different Load Step are set to 2, then the penetration will progress at 4 cm/s. A detailed description of how these multipliers work can be found in section 7.3 of the [Tutorial manual 2022](https://github.com/Anura3D/Anura3D_OpenSource/blob/main/manuals/TutorialManual_2022.pdf).

**Reply by harryfor4** [reply]
Thank you for the detailed explanatory comment.

**Reply by murphy-john** [reply]
Great addition Alba! I forgot about this option!
